### PR TITLE
Server overrides fix

### PR DESCRIPTION
**Pull request by joeldenning**
This expands server override capabilities to be able to control whether the server or browser injects the import map. Also, full refactor of the docs.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import-map-overrides has two primary configuration options:
+
+1. [Import Map Type](#import-map-type)
+2. [Override Mode](#override-mode)
+
+## Import Map Type
+
+By default, import map overrides will assume you are working with native import maps. However, other import map polyfills (such as SystemJS) can also be used with import map overrides.
+
+If using an import map polyfill, you must indicate what kind of import map you are setting overrides for. You do this by inserting you must add a `<meta>`
```
Comment (frehner) [inline docs/configuration.md]: `You do this by inserting you must add`

File: docs/configuration.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+1. Native import maps are only supported in Chrome@>=76 under the _Experimental Web Platform Features_ flag. Only one import map (including the import-map-overrides map) can be on a web page at a time when using native import maps. ([Details](https://github.com/WICG/import-maps/issues/199)).
+
+## Override Mode
+
+To support a variety of use cases, import map overrides has four "override modes" that control how and whether import-map-overrides inserts import maps into the DOM:
+
+1. [Client-side multiple maps](client-side-multiple-maps) (default)
+1. [Client-side single map](client-side-single-map)
+1. [Server-side multiple maps](server-side-multiple-maps)
+1. [Server-side single map](server-side-single-map)
```
Comment (frehner) [inline docs/configuration.md]: do these links work? or do they need a `#` in front of them?

**Comment by joeldenning** [comment]
👍 thanks - updated

File: docs/configuration.md (hunk 2 above)
Comment (joeldenning) [reply docs/configuration.md]: nope, i forgot the hash. Updated

**Pull request merged**